src/nnet3/nnet-am-decodable-simple.h

```diff
@@ -392,6 +392,170 @@ class DecodableAmNnetSimpleParallel: public DecodableInterface {
+  bool GetFinishedUtterance(std::string *uttid, Matrix<BaseFloat> *output_matrix);
+
+  // This is a blocking call. When we call this function, we try to finish
+  // computing all the utterances which are still stored in this class.
+  // If 'flush == true', it would ensure that even if a batch wasn't ready,
+  // the computation would be run. When all the feature matrixes are read,
+  // we always call it one time to deal with the remain.
+  // This function is also useful in overload case.
+  void Compute(bool flush);
+
+ private:
+  KALDI_DISALLOW_COPY_AND_ASSIGN(NnetSimpleBatch);
+
+  // When an utterance is taken by GetFinishedUtterance(), clear its information.
+  void Clear(std::string utt_id);
+
+  // According to the information in batch_info_queue_, prepare the 'batch' data,
+  // ComputationRequest , compute and get the results out.
+  void DoNnetComputation();
+
+  // Gets the iVector that will be used for this chunk of frames, if we are
+  // using iVectors (else does nothing).  note: the num_output_frames is
+  // interpreted as the number of t value, which in the subsampled case is not
+  // the same as the number of subsampled frames (it would be larger by
+  // opts_.frame_subsampling_factor).
+  void GetCurrentIvector(std::string utt_id,
+                         int32 output_t_start,
+                         int32 num_output_frames,
+                         Vector<BaseFloat> *ivector);
+
+  // called from constructor
+  void CheckAndFixConfigs();
+
+  // called from AcceptInput()
+  void CheckInput(const std::string &utt_id,
+                  const MatrixBase<BaseFloat> &feats,
+                  const VectorBase<BaseFloat> *ivector = NULL,
+                  const MatrixBase<BaseFloat> *online_ivectors = NULL);
+
+  // called from AcceptInput() or Compute(). Prepare the batch_info_queue_ which
+  // will be used to compute.
+  void PrepareBatchInfo();
+
+
+  NnetSimpleComputationOptions opts_;
+  const Nnet &nnet_;
+  int32 nnet_left_context_;
+  int32 nnet_right_context_;
+  int32 output_dim_;
+  // the log priors (or the empty vector if the priors are not set in the model)
+  CuVector<BaseFloat> log_priors_;
+
+  std::unordered_map<std::string, Matrix<BaseFloat>> feats_;
+
+  // ivector_ is the iVector if we're using iVectors that are estimated in batch
+  // mode.
+  std::unordered_map<std::string, Vector<BaseFloat>> ivectors_;
+
+  // online_ivector_feats_ is the iVectors if we're using online-estimated ones.
+  std::unordered_map<std::string, Matrix<BaseFloat>> online_ivector_feats_;
+
+  // online_ivector_period_ helps us interpret online_ivector_feats_; it's the
+  // number of frames the rows of ivector_feats are separated by.
+  int32 online_ivector_period_;
+
+  // a reference to a compiler passed in via the constructor, which may be
+  // declared at the top level of the program so that we don't have to recompile
+  // computations each time.
+  CachingOptimizingCompiler &compiler_;
```

Comment:
For google-style-guide reaons, please make this a pointer not a reference.  (It's non-const).